azure-core/azure/core/pipeline/policies/credentials_async.py

```diff
@@ -1,41 +1,27 @@
-# --------------------------------------------------------------------------
-#
+# -------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
-#
-# The MIT License (MIT)
-#
-# Permission is hereby granted, free of charge, to any person obtaining a copy
-# of this software and associated documentation files (the ""Software""), to
-# deal in the Software without restriction, including without limitation the
-# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
-# sell copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-#
-# The above copyright notice and this permission notice shall be included in
-# all copies or substantial portions of the Software.
-#
-# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
-# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
-# IN THE SOFTWARE.
-#
+# Licensed under the MIT License. See LICENSE.txt in the project root for
+# license information.
 # --------------------------------------------------------------------------
-import asyncio
-from collections.abc import AsyncIterator
-import functools
-import logging
-from typing import Any, Callable, Optional, AsyncIterator as AsyncIteratorType
-
-import requests
-from requests.models import CONTENT_CHUNK_SIZE
+from typing import Any, Iterable, Mapping, TYPE_CHECKING  # pylint:disable=unused-import
 
+from azure.core.pipeline import PipelineRequest, PipelineResponse
 from azure.core.pipeline.policies import AsyncHTTPPolicy
 
+if TYPE_CHECKING:
+    from azure.core.credentials import SupportsGetToken  # pylint:disable=unused-import
+
+
+class AsyncBearerTokenCredentialPolicy(AsyncHTTPPolicy):  # pylint:disable=too-few-public-methods
+    """Adds a bearer token Authorization header to requests."""
+
+    def __init__(self, credentials: "SupportsGetToken", scopes: Iterable[str], **kwargs: Mapping[str, Any]) -> None:
+        # pylint:disable=unused-argument
+        super(AsyncBearerTokenCredentialPolicy, self).__init__()
```

Comment:
Why aren't `kwargs` forwarded here? Will the base classes behave badly?

Reply:
`Async/HTTPPolicy` constructors [don't accept `kwargs`](https://github.com/Azure/azure-sdk-for-python/pull/4685/files#diff-c5bc1f6826a50e170a283580db4eca27R44). Can't speak to the rationale there.

Reply:
@johanste can you comment? I'd normally expect every initializer in a cooperative inheritance situation to  pick off the arguments they actually care about by naming them, but accept `**kwargs` to pass on. Otherwise mixins become sensitive to the order they are listed in the class def.